File: python/DUELink/Image.py
class ImageController:
    def __init__(self, width: int, height: int, data):
        self.Width = width
        self.Height = height
        self.Data = [0] * ((width * height) + 2)

        self.Data[0] = self.Width
        self.Data[1] = self.Height

        for i in range (width * height):
            self.Data[2 + i] = data[i]

File: dotnet/GHIElectronics.DUELink/Image.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GHIElectronics.DUELink {
    public class Image {
        public uint[] Data { get; }
        public uint Width { get; }
        public uint Height { get;  }
        public Image(uint width, uint height, uint[] data) {
            if (width == 0 || height == 0 || data == null || data.Length  < (width * height)) {
                throw new ArgumentException("Invalid argument.");
            }
            this.Width = width;
            this.Height = height;
            this.Data = new uint[width * height + 2];

            this.Data[0] = this.Width;
            this.Data[1] = this.Height;

            Array.Copy(data, 0, this.Data, 2, this.Width * this.Height);
        }


    }
}
